smack-extensions/src/test/java/org/jivesoftware/smackx/usertune/UserTuneElementTest.java

```diff
@@ -0,0 +1,88 @@
+/**
+ *
+ * Copyright 2019 Aditya Borikar.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.smackx.usertune;
+
+import static junit.framework.TestCase.assertEquals;
+import static org.jivesoftware.smack.test.util.XmlUnitUtils.assertXmlSimilar;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+
+import org.jivesoftware.smack.parsing.SmackParsingException;
+import org.jivesoftware.smack.test.util.SmackTestSuite;
+import org.jivesoftware.smack.test.util.TestUtils;
+import org.jivesoftware.smack.xml.XmlPullParser;
+import org.jivesoftware.smack.xml.XmlPullParserException;
+import org.jivesoftware.smackx.usertune.element.UserTuneElement;
+import org.jivesoftware.smackx.usertune.provider.UserTuneProvider;
+
+import org.junit.Test;
+
+public class UserTuneElementTest extends SmackTestSuite {
+
+    @Test
+    public void toXmlTest() throws IOException, XmlPullParserException, SmackParsingException {
+        String xml = "<tune xmlns='http://jabber.org/protocol/tune'>" +
+                "<artist>Yes</artist>" +
+                "<length>686</length>" +
+                "<rating>8</rating>" +
+                "<source>Yessongs</source>" +
+                "<title>Heart of the Sunrise</title>" +
+                "<track>3</track>" +
+                "<uri>http://www.yesworld.com/lyrics/Fragile.html#9</uri>" +
+                "</tune>";
+
+        URI uri = null;
+        try {
+            uri = new URI("http://www.yesworld.com/lyrics/Fragile.html#9");
+        } catch (URISyntaxException e) {
+            throw new IllegalArgumentException("Invalid URI");
+        }
+
+        UserTuneElement.Builder builder = UserTuneElement.getBuilder();
+        UserTuneElement userTuneElement = builder.setArtist("Yes")
+                .setLength(686)
+                .setRating(8)
+                .setSource("Yessongs")
+                .setTitle("Heart of the Sunrise")
+                .setTrack(3)
+                .setUri(uri)
+                .build();
+        assertXmlSimilar(xml, userTuneElement.toXML().toString());
+        XmlPullParser parser = TestUtils.getParser(xml);
+        UserTuneElement parsed = UserTuneProvider.INSTANCE.parse(parser);
+        assertEquals(xml, parsed.toXML().toString());
+    }
+
+    @Test(expected = XmlPullParserException.class)
+    public void unknownUserTuneValueExceptionTest() throws Exception {
```

Comment:
Please avoid `throws Exception`, instead explicitly state the checked Exceptions this is potentially going to throw.